src/Features/CSharp/Portable/RemoveUnreachableCode/CSharpRemoveUnreachableCodeCodeFixProvider.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.RemoveUnreachableCode
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.RemoveUnreachableCode), Shared]
+    internal class CSharpRemoveUnreachableCodeCodeFixProvider : SyntaxEditorBasedCodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+            ImmutableArray.Create(IDEDiagnosticIds.RemoveUnreachableCodeDiagnosticId);
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var diagnostic = context.Diagnostics[0];
+            context.RegisterCodeFix(new MyCodeAction(
+                FeaturesResources.Remove_unreachable_code,
+                c => FixAsync(context.Document, diagnostic, c)), diagnostic);
+
+            return SpecializedTasks.EmptyTask;
+        }
+        
+
+        protected override bool IncludeDiagnosticDuringFixAll(Diagnostic diagnostic)
+            => !diagnostic.Properties.ContainsKey(CSharpRemoveUnreachableCodeDiagnosticAnalyzer.IsCascadedSection);
+
+        protected override Task FixAllAsync(
+            Document document, 
+            ImmutableArray<Diagnostic> diagnostics, 
+            SyntaxEditor editor, 
+            CancellationToken cancellationToken)
+        {
+            var syntaxRoot = editor.OriginalRoot;
+
+            foreach (var diagnostic in diagnostics)
+            {
+                var firstUnreachableStatementLocation = diagnostic.AdditionalLocations.Single();
```

Comment:
@CyrusNajmabadi The diagnostic reported for the child of the `if` statement does not have additional locations. It's an early-out edge case for the parent not being a block or switch section.